presto-main/src/test/java/com/facebook/presto/connector/system/TestSystemSession.java

```diff
@@ -0,0 +1,78 @@
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.system;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.common.transaction.TransactionId;
+import com.facebook.presto.common.type.VarcharType;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.facebook.presto.spi.security.ConnectorIdentity;
+import com.facebook.presto.spi.security.SelectedRole;
+import com.facebook.presto.spi.session.PropertyMetadata;
+import com.facebook.presto.testing.TestingConnectorSession;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import java.security.Principal;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+public class TestSystemSession
+{
+    @Test
+    public void testSystemSessionForExtraCredentials()
+    {
+        TransactionId transactionId = new TransactionId(UUID.randomUUID());
+
+        ConnectorTransactionHandle transactionHandle = new GlobalSystemTransactionHandle("connector-id", transactionId);
+
+        Optional<Principal> principal = Optional.of(() -> "principal");
+
+        Optional<SelectedRole> selectedRole = Optional.empty();
+
+        ConnectorIdentity connectorIdentity = new ConnectorIdentity("user", principal, selectedRole) {
+            @Override
+            public Map<String, String> getExtraCredentials()
+            {
+                Map<String, String> extraCredentials = new HashMap<>();
+                extraCredentials.put("key1", "value1");
+                extraCredentials.put("key2", "value2");
+                return extraCredentials;
+            }
+        };
+
+        TestingConnectorSession testingConnectorSession = new TestingConnectorSession(connectorIdentity,
+                ImmutableList.of(new PropertyMetadata<>("property-name",
+                        "This is a description",
+                        VarcharType.createUnboundedVarcharType(),
+                        String.class,
+                        "defaultValue",
+                        false,
+                        Object::toString,
+                        value -> value)));
+
+        Session resultSession = SystemConnectorSessionUtil.toSession(transactionHandle, testingConnectorSession);
+
+        assertNotNull(resultSession, "Session should not be null");
+        assertEquals(resultSession.getIdentity().getUser(), "user", "User should match");
+        assertEquals(resultSession.getIdentity().getPrincipal().get().getName(), "principal", "Principal should match");
+        assertEquals(resultSession.getIdentity().getExtraCredentials().get("key1"), "value1", "Extra credential for key1 should match");
+        assertEquals(resultSession.getIdentity().getExtraCredentials().get("key2"), "value2", "Extra credential for key2 should match");
```

Comment:
Just remove these messages, they are not helpful because they don't tell you the mismatched values.